database/layer/admin-permission.js

```diff
@@ -1,50 +1,71 @@
-const { PGPool } = require('./pg_pool')
-const pool = new PGPool()
 const { usherDb } = require('./knex')
 const { pgErrorHandler } = require('../utils/pgErrorHandler')
 
-const insertPermissionByClientId = async (clientId, permissionname, permissiondescription) => {
-  const sql = 'INSERT INTO usher.permissions (clientkey, name, description) SELECT key, $1, $2 FROM usher.clients WHERE client_id = $3'
-  const sqlParams = [permissionname, permissiondescription, clientId]
+/**
+ * Insert a new permission for a specific client
+ * @param {string} clientId - The client ID
+ * @param {string} permissionName - The name of the permission
+ * @param {string} permissionDescription - A description of the permission
+ * @returns {Promise<Object>} - A promise that resolves to the inserted permission object
+ */
+const insertPermissionByClientId = async (clientId, permissionName, permissionDescription) => {
   try {
-    await pool.query(sql, sqlParams)
-    return 'Insert successful'
-  } catch (error) {
-    if (error.message === 'duplicate key value violates unique constraint "permissions_name_clientkey_uq"') {
-      const errClientPermissionAlreadyExists = `A permission ${permissionname} already exists matching client_id`
-      return `Insert failed: ${errClientPermissionAlreadyExists}`
+    const client = await usherDb('clients').where({ client_id: clientId }).first('key')
+    if (!client) {
+      throw new Error(`Client with id ${clientId} not found`)
     }
-    return `Insert failed: ${error.message}`
+
+    const [permission] = await usherDb('permissions')
+      .insert({
+        clientkey: client.key,
+        name: permissionName,
+        description: permissionDescription
+      })
+      .returning('*')
+
+    return permission
+  } catch (error) {
+    throw pgErrorHandler(error)
   }
 }
 
-const updatePermissionByPermissionname = async (clientId, permissionname, permissiondescription) => {
-  const sql = 'UPDATE usher.permissions p SET description = $1 WHERE EXISTS (SELECT 1 FROM usher.clients c WHERE c.client_id = $2) AND p.name = $3'
-  const sqlParams = [permissiondescription, clientId, permissionname]
+const updatePermissionByPermissionName = async (clientId, permissionName, permissionDescription) => {
```

Comment:
Perhaps we can remove these functions in the future and rely on generic update functions instead.
```suggestion
const updatePermissionDescription = async (clientId, permissionName, permissionDescription) => {
```

Reply:
Yes, let's focus on `PGPool` in this PR. I just wanted to highlight the issue so we can fix later.